src/NuGet.Core/NuGet.Packaging/Rules/NoRefOrLibFolderInPackageRule.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    class NoRefOrLibFolderInPackageRule : IPackageRule
```

Comment:
Nit: consider making the class's visibility explicit

```suggestion
    internal class NoRefOrLibFolderInPackageRule : IPackageRule
```